tests/integration_tests/functional/test_balloon.py

```diff
@@ -14,18 +14,40 @@
 from framework.artifacts import ArtifactCollection, ArtifactSet
 from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
 from framework.matrix import TestMatrix, TestContext
-from framework.utils import get_free_mem_ssh
+from framework.utils import get_free_mem_ssh, run_cmd
 
 import host_tools.network as net_tools  # pylint: disable=import-error
 
 
 MB_TO_PAGES = 256
 
 
-def get_rss_mem_by_pid(pid):
-    """Get the RSS memory that a guest uses, given the pid of the guest."""
-    output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
-    return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+def get_stable_rss_mem_by_pid(pid, percentage_delta=0.5, timeout_secs=30):
+    """
+    Get the RSS memory that a guest uses, given the pid of the guest.
+
+    Wait till the fluctuations in RSS drop below percentage_delta. If timeout
+    is reached before the fluctuations drop, raise an exception.
+    """
+
+    def get_rss_from_pmap():
+        _, output, _ = run_cmd("pmap -X {}".format(pid))
+        return int(output.split('\n')[-2].split()[1], 10)
+
+    rss = get_rss_from_pmap()
+    delta = 100
+    while delta > percentage_delta and timeout_secs > 0:
+        prev_rss = rss
+        time.sleep(1)
+
+        rss = get_rss_from_pmap()
+        timeout_secs -= 1
+        delta = (abs(prev_rss - rss)/float(prev_rss)) * 100
+
+    if timeout_secs == 0:
```

Comment:
The `if` condition should be in the `while` loop.

Reply:
It doesn't have to be right? It will exit the while loop if timeout_secs reaches 0. Let me know or maybe I am missing something?

Reply:
Ah, sorry for the vague comment there :)

You can simplify the while loop condition and check `timeout_secs` only once inside the loop.

Reply:
Oh yes, that's a good idea, I didn't catch that from the comment, updated it.